### PR TITLE
Jennyf/remove pii

### PR DESCRIPTION
**Pull request by jennyf19**
Initial commit for #492 

- Went through all logs and found two that now contain pii (authority, client_id, redirect_uri), as per gdpr spec. Have moved those to LogPii and added in non-pii logs. 
- Doing smaller PRs for easier and clearer review.

### CODE REVIEW

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs [hunk 1]
```diff
@@ -113,30 +113,49 @@ internal class AuthenticationRequestParameters
...
-            if(IsClientCredentialRequest)
+            // Create Pii enabled string builder
+            if (Logger.PiiLoggingEnabled)
             {
-                builder.AppendLine("Client Certificate Provided? - " + (ClientCredential.Certificate != null));
+                StringBuilder builder = new StringBuilder(Environment.NewLine + "=== Request Data ===" +
+                                                          Environment.NewLine +
+                                                          "Authority Provided? - " + (Authority != null) +
+                                                          Environment.NewLine);
+                builder.AppendLine("Client Id - " + ClientId);
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: whole blob is not PII. you can selectively remove information that is qualified to be PII. this is a critical piece of information that we need

Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: I removed the pii info for the non-pii log.

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs [hunk 2]
```diff
@@ -130,13 +132,32 @@ public void LogState()
...
 
+            // Create no Pii enabled string builder
+            builder = new StringBuilder(Environment.NewLine + "=== Request Data ===" +
+                                                      Environment.NewLine);
+            builder.AppendLine("Scopes - " + Scope?.AsSingleString());
+            builder.AppendLine("Validate Authority? - " + ValidateAuthority);
+            builder.AppendLine("LoginHint provided? - " + !string.IsNullOrEmpty(LoginHint));
+            dict = MsalHelpers.ParseKeyValueList(ExtraQueryParameters, '&', true, RequestContext);
+            builder.AppendLine("Extra Query Params Keys (space separated) - " + dict.Keys.AsSingleString());
+            dict = MsalHelpers.ParseKeyValueList(ExtraQueryParameters, '&', true, RequestContext);
```
Comment (SomkaPe) [inline src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: what is the point of doing this twice ?

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs [hunk 3]
```diff
@@ -130,13 +132,32 @@ public void LogState()
...
+            builder.AppendLine("Slice Parameters Keys(space separated) - " + dict.Keys.AsSingleString());
```
Comment (SomkaPe) [inline src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: same data are logged as "Extra Query Params Keys"

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs [hunk 4]
```diff
@@ -130,13 +132,32 @@ public void LogState()
...
+            if (IsClientCredentialRequest)
             {
                 builder.AppendLine("Client Certificate Provided? - " + (ClientCredential.Certificate != null));
             }
 #endif
-            RequestContext.Logger.Info(builder.ToString());
+            RequestContext.Logger.InfoPii(builder.ToString());
 
+            // Create no Pii enabled string builder
+            builder = new StringBuilder(Environment.NewLine + "=== Request Data ===" +
```
Comment (brentschmaltz) [inline src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: why have this on two lines?

Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: not sure...fixed it...thanks

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs (hunk 2 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: good catch. fixed

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs (hunk 3 above)
Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: removed for pii.

File: src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs [hunk 5]
```diff
@@ -63,13 +63,19 @@ protected RequestBase(AuthenticationRequestParameters authenticationRequestParam
...
+            // Log contains Pii 
+            authenticationRequestParameters.RequestContext.Logger.InfoPii(string.Format(CultureInfo.InvariantCulture,
                 "=== Token Acquisition ({4}) started:\n\tAuthority: {0}\n\tScope: {1}\n\tClientId: {2}\n\tCache Provided: {3}",
                 authenticationRequestParameters?.Authority?.CanonicalAuthority,
                 authenticationRequestParameters.Scope.AsSingleString(),
                 authenticationRequestParameters.ClientId,
                 TokenCache != null, this.GetType().Name));
 
+            // Log does not contain Pii
+            authenticationRequestParameters.RequestContext.Logger.Info(string.Format(CultureInfo.InvariantCulture,
```
Comment (SomkaPe) [inline src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs]: We might consider logging authority if it is whitelisted (well known public authorities, not oii) as no pii,

Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs]: We can log the trusted host. I will add a check for this.

**Comment by SomkaPe** [comment]
:shipit:

File: src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs [hunk 6]
```diff
@@ -125,18 +128,34 @@ public void LogState()
...
+            // Create no Pii enabled string builder
+            builder = new StringBuilder(Environment.NewLine + "=== Request Data ===" + Environment.NewLine +
+                                        "Authority Provided? - " + (Authority != null) + Environment.NewLine);
+            builder.AppendLine("Scopes - " + Scope?.AsSingleString());
+            builder.AppendLine("Validate Authority? - " + ValidateAuthority);
+            builder.AppendLine("LoginHint provided? - " + !string.IsNullOrEmpty(LoginHint));
+            builder.AppendLine("User provided? - " + (User != null));
             dict = MsalHelpers.ParseKeyValueList(ExtraQueryParameters, '&', true, RequestContext);
-            builder.AppendLine("Slice Parameters Keys(space separated) - " + dict.Keys.AsSingleString());
+            builder.AppendLine("Extra Query Params Keys (space separated) - " + dict.Keys.AsSingleString());
```
Comment (kpanwar) [inline src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: why are you adding same log in again?

Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: fixed.

Comment (jennyf19) [reply src/Microsoft.Identity.Client/Internal/Requests/AuthenticationRequestParameters.cs]: I added in the slice parameters, as per our conversation

**Pull request merged**